### PR TITLE
fix(server): Register services known versions

### PR DESCRIPTION
**Pull request by rldhont**
## Description

In QGIS Server only default supported OGC services versions has been registered :

* WMS 1.3.0 even if 1.1.1 is supported
* WFS 1.1.0 even if 1.0.0 is supported

### CODE REVIEW

**Comment by rldhont** [comment]
@dmarteau can you take a look at these changes

**Comment by dmarteau** [comment]
Seems fine to me; 

This will also prevent overriding 1.1.0 with python implementation (native implementation is always preferred for given version).

**Comment by rldhont** [comment]
Not staled, just waiting for reviews.

File: src/server/services/wms/qgswmsutils.h [hunk 1]
```diff
@@ -48,6 +48,11 @@ namespace QgsWms
     WEBP, //!< WEBP format
   };
 
+  /**
+   * Returns the highest version supported by this implementation
```
Comment (pblottiere) [inline src/server/services/wms/qgswmsutils.h:52]: Can you add `\since` info please?

File: src/server/services/wfs/qgswfs.cpp [hunk 2]
```diff
@@ -47,14 +47,16 @@ namespace QgsWfs
 
       /**
        * Constructor for WFS service.
+       * \param version Version of the WFS service.
```
Comment (lbartoletti) [inline src/server/services/wfs/qgswfs.cpp:50]:
```suggestion
       * \param version Version of the WFS service. \since QGIS 3.28
```

Comment (rldhont) [reply src/server/services/wfs/qgswfs.cpp:50]: Since 3.28.0 or 3.22.12 ?

File: src/server/services/wms/qgswmsutils.h (hunk 1 above)
Comment (rldhont) [reply src/server/services/wms/qgswmsutils.h:52]: Since 3.28.0 or 3.22.12 ?

Comment (pblottiere) [reply src/server/services/wms/qgswmsutils.h:52]: 3.22.12

File: src/server/services/wfs/qgswfs.cpp (hunk 2 above)
Comment (lbartoletti) [reply src/server/services/wfs/qgswfs.cpp:50]: 3.22.12 since it's backported.

**Comment by rldhont** [comment]
Thansk @pblottiere and @lbartoletti for your review. I have updated the code by adding `\since`

File: src/server/services/wfs/qgswfs.cpp [hunk 3]
```diff
@@ -47,14 +47,16 @@ namespace QgsWfs
 
       /**
        * Constructor for WFS service.
+       * \param version Version of the WFS service.  \since QGIS 3.22.12
```
Comment (pblottiere) [inline src/server/services/wfs/qgswfs.cpp:50]: `\since` is strangely positioned no?

Comment (rldhont) [reply src/server/services/wfs/qgswfs.cpp:50]: I do it like @lbartoletti proposed it.

Comment (rldhont) [reply src/server/services/wfs/qgswfs.cpp:50]: @lbartoletti is the `\since` has to be for the method or for the parameter ?

Comment (lbartoletti) [reply src/server/services/wfs/qgswfs.cpp:50]: For the parameter, but it should be `(since xx)` and not `\since xx`, sorry

Comment (rldhont) [reply src/server/services/wfs/qgswfs.cpp]: @lbartoletti @pblottiere changes done. Thanks

**Comment by pblottiere** [comment]
Hi @rldhont,

I think that https://github.com/qgis/QGIS/pull/49484/files#r958092282 is not really resolved (`\since` is still missing for this method).

**Comment by rldhont** [comment]
> Hi @rldhont,
> 
> I think that https://github.com/qgis/QGIS/pull/49484/files#r958092282 is not really resolved (`\since` is still missing for this method).

Hi @pblottiere, I was sure to commit and push the changes for this method. Sorry.

**Comment by rldhont** [comment]
All the `since` has been added.

**Review by pblottiere — APPROVED** [review]

**Comment by rldhont** [comment]
Thanks @pblottiere

**Pull request merged**